Added Context, Container and Componemt Diagrams

kogoe1 opened the pull request.

hugh-emerson commented:
Your context diagram is good.

The component/container diagram is confused.
- First of all, why is there a single diagram labelled "container and component"? They should be separate diagrams. What you have looks like it's meant to be a container diagram. Did you forget to save the component diagram?
- You have a box labelled "Single Page application", which is not what you have. That is a specific term that refers to a webapp where the browser loads a static, javascript-heavy page that then makes API calls to load dynamic content/enable interaction.
- If the "web application" is the Flask app which responds to the browser's request to load the page, why isn't there an arrow to it from the user?
- How do your Docker container and gunicorn/flask fit into this?

For a component diagram, you zoom in on the Flask app, breaking it down into sections of the codebase, i.e. templates/models/controller/TrelloUtility

kogoe1 commented:
Thanks for the feedback Hugh.
My bad for not clarifying that there are two pages in the "Container and component diagrams" file. Page 1 has Container diagram and page 2 the component diagram. You can toggle between the pages at the bottom of the file (will rename the page names appropriately).

I'll wait for your feedback on the component diagram and make all necessary changes in one go

kogoe1 commented:
Hi Hugh - thought to let you know that I have split the container and component diagrams into two separate files and have also modified them.
The pull request for the change is here...
https://github.com/kogoe1/DevOps-Course-Starter/pull/8
